leveldbstatic/raw.nim

```diff
@@ -15,16 +16,23 @@ when defined(posix):
   {.passc: "-DLEVELDB_PLATFORM_POSIX".}
 
 
-{.passc: "-DHAVE_FDATASYNC=0".}
-{.passc: "-DHAVE_FULLFSYNC=0".}
-{.passc: "-DHAVE_O_CLOEXEC=0".}
-{.passc: "-DHAVE_CRC32C=0".}
-{.passc: "-DHAVE_SNAPPY=0".}
-{.passc: "-DHAVE_ZSTD=0".}
-{.passc: "-DHAVE_Zstd=0".}
+static:
+  proc doesCompile(cfile: string): int =
```

Comment:
Unfortunately this will break if the system uses clang or a non-standard gcc, etc.

It'd probably be better to run and parse the cmake output and grep for these flags. I forget the command, but cmake does produce cached outputs of what flags are enabled. You can basically read it and grep for the results. However in doing that would really only make sense if `cmake` is re-run each time at compile time. That complicates things too. :)

It might make sense to just vendor and compile these libs as well. Basically go for the "fat" version of the lib! Though I think `fdatasync` might be OS specific.

Yet another option is to use something like `pkg-config` to check for the libraries.

Oh for `fullfsync` might consider using direct `when defined(linux) or defined(macosx): {.passc: ....}`. You could combine them with custom Nim defines as well, like `nim -d:leveldbFullSync` to allow user selection.

I'd probably go with vendored snappy, zstd, and the rest as enabled based on Nim defines and/or platform selects.